### PR TITLE
Reduce binary sizes

### PR DESCRIPTION
**Pull request by mauromorales**
relates to: kairos-io/kairos#2285

### CODE REVIEW

**Review by Itxaka — APPROVED** [review]
No immucore, k9s, k3s etc all?

**Comment by Itxaka** [comment]
Btw I think kceypt is not being shipped as part of Kairos anymore as everything is done via it's public apis directly on immucore and agent

**Comment by mauromorales** [comment]
@Itxaka good point, will add those

**Comment by Itxaka** [comment]
> @Itxaka good point, will add those

Im a bit worried about immucore as it runs on an empty system no idea what upx needs to be able to autodecompress?

File: packages/toolchain-go/build.yaml [hunk 1]
```diff
@@ -1,11 +1,19 @@
...
+image: "golang:{{.Values.tag}}"
 {{ end }}
 
-{{ if eq .Values.variant "alpine" }}
 prelude:
+{{ if eq .Values.variant "alpine" }}
 - apk update
 - apk add curl make git bash
+{{ end }}
+{{ if eq .Values.variant "bookworm" }}
```
Comment (mudler) [inline packages/toolchain-go/build.yaml:12]: :+1: 

unbelievelable, only bookworm doesn't have upx. It is such an historic package - I'm a bit shocked :)

https://packages.debian.org/search?searchon=names&keywords=upx

**Review by mudler — APPROVED** [review]

File: packages/toolchain-go/build.yaml (hunk 1 above)
Comment (mauromorales) [reply packages/toolchain-go/build.yaml:12]: yup I was a bit confused there

File: packages/toolchain-go/build.yaml [hunk 2]
```diff
@@ -1,11 +1,19 @@
...
+- apt-get update && apt-get install -y curl xz-utils
+- curl -LO https://github.com/upx/upx/releases/download/v4.2.2/upx-4.2.2-amd64_linux.tar.xz
```
Comment (mudler) [inline packages/toolchain-go/build.yaml:14]: of course this doesn't work for arm64, need to handle the architecture properly here

**Comment by mudler** [comment]
aren't we missing here to compress/minimize immucore too?

**Comment by mauromorales** [comment]
@Itxaka ok I'll do immucore last and in a different PR

**Comment by mauromorales** [comment]
@mudler yes my bad, will add a bunch of others I completely forgot we had there. Also, will remove the docs since they account for half of the size of the compressed binary (12Mb and 24 total)

**Pull request merged**